### PR TITLE
Adds a bloodtest for changelings.

### PR DESCRIPTION
**Pull request by nemvar**
## About The Pull Request

This adds a blood test to changelings. Just like this entire antag, it is a reference to "The Thing". Basically, heating up the blood of a changeling will result in it displaying abnormal properties and will allow you to identify a changeling easily. Changelings gain the ability to identify other changelings by ingesting their blood.

This also adds an ability to changelings that supresses the production of chemicals, but makes their blood immune to the blood test. Deactivating this ability takes a long time, although respecing your abilities cancels it instantly. This ability allows changeling that try to take a more stealthy approach the possiblity to hide their true nature, even if shitcurity instates mandatory blood inspection.

## Why It's Good For The Game

I'm not completly convinced this is good for the game. That's why this is a draft for now. But I do feel like the lack of a hard tell for lings encourages bad gameplay. Lings are also horrendously strong so bringing them down a notch won't be all that bad.

## Changelog
:cl:
add: Adds a blood test for lings. Heating their blood will reveal their true identity.
add: A new ability for lings: Hide identity. You won't regenerate chemicals while this is active but your blood will resist the blood test. Turning this ability off takes a long time.
/:cl:

### CODE REVIEW

**Comment by nicbn** [comment]
I'm against having a way to test lings: the only thing that should denounce one of them is their own lack of stealth

**Comment by terranaut1** [comment]
I've seen this for a while now while playing on other servers. For those servers, it works very well - but that might be partially related to the generally slower pace of play there. I like the ability a lot, I mostly main security there and I find that getting someone's blood checked never truly makes me write them off as a changeling, instead it makes me more paranoid and I try to juggle keeping an eye on whoever while being on guard for all other lings.

**Comment by Tlaltecuhtli** [comment]
i prefer giving the suspect an n2o/bz air tank and wait for the "bwoink" noise

**Comment by Akrilla** [comment]
Time to rush the tempature gun so I can powergame lings

**Comment by nemvar** [comment]
Due to the way blood works, this won't actually trigger if it's still inside the ling.

**Comment by knacker48** [comment]
When you say you adding an ability is it one that you need to purchase like biodegrade or is it one of the starting one that you always have?

**Comment by nemvar** [comment]
Purchasable for 1 DNA point.

**Comment by Burns-The-Potatos** [comment]
So basically sec can walk down a hall, stun some random grey, and test blood right in front of them, exposing them as a changeling in 10 seconds? Pretty lame, in my opinion.

**Comment by Fox-McCloud** [comment]
Having had this over at Paradise for a while, I can tell you what it leads to.

The moment there's even *remotely* a suspicion of a changeling (there doesn't even have to be one; it just takes one terrible officer/command member to push for it), security begin systematically stopping and forcibly taking the blood of every single member on station, one by one. If you so much as refuse, no matter you're reason, you're stunned cuffed, detained on the spot and they take your blood anyway.

In terms of meta-behavior, this is by the far the worst we ever experienced.

Don't try it here; I guarantee it'll lead to the same thing.

**Comment by PKPenguin321** [comment]
Fox's post is enough for me to -1 this, even though it sounds cool in theory

**Comment by nero1024** [comment]
How about have the discovery chance be based on how many loud abilities (stuff like armblades, armor, etc.) the changeling has?

**Comment by 81Denton** [comment]
I'm a fan of The Thing and this sounds good in theory, but with our playerbase it'll go down just like how Fox described it.

Just imagine Sybil sec detaining everyone they can get their hands on to obtain blood samples for valids, yikes.

**Comment by 4dplanner** [comment]
I mean, that's basically already how hiveminds goes, and it's terrible. It'd be even worse without even needing to buy implants.

**Comment by YakumoChen** [comment]
:-1: We disapprove antag checks for a reason

**Comment by Tlaltecuhtli** [comment]
ok idea: the ling test recipe is blood + potassium + water, if it fails it blows you up

File: code/modules/antagonists/changeling/changeling.dm [hunk 1]
```diff
@@ -576,3 +576,9 @@
 
 /datum/antagonist/changeling/xenobio/antag_listing_name()
 	return ..() + "(Xenobio)"
+
+/datum/proc/changelingTest(datum/reagents/holder ,datum/reagent/blood/B, temperature)
```
Comment (JJRcop) [inline code/modules/antagonists/changeling/changeling.dm]: Whoops this proc is on `/datum`

**Comment by Shadowflame909** [comment]
Have the Ling-Test be skewium instead.

Nobody likes Skewium, not even lings.

**Comment by nemvar** [comment]
Closing this because not only is the idea probably not good for the game, but the code for it is also somewhat shit.

**Pull request closed without merging**